lib/nerve/rate_limiter.rb

```diff
@@ -0,0 +1,45 @@
+module Nerve
+  # RateLimiter implements a TokenBucket-based rate-limiting strategy.
+  # It allows an average of `average_rate` tokens per `period`, with a
+  # maximum burst of `max_burst` when the bucket is full.
+  # `period` is the time period for `average_rate`, in seconds.
+  # Note: a single instance of RateLimiter is *not* thread-safe.
+  # See: https://en.wikipedia.org/wiki/Token_bucket
+  class RateLimiter
+    def initialize(average_rate: 1, max_burst: 10, period: 1)
+      raise TypeError, "average_rate should be numeric" unless average_rate.is_a? Numeric
+      raise TypeError, "max_burst should be numeric" unless max_burst.is_a? Numeric
+      raise TypeError, "period should be numeric" unless period.is_a? Numeric
```

Comment:
can we consolidate average_rate and period?  average_rate can be float to express rate sub-second?

Reply:
for example https://godoc.org/golang.org/x/time/rate#Limit